src/zoo.js

```diff
@@ -11,57 +11,137 @@ eslint no-unused-vars: [
 
 const data = require('./data');
 
-function animalsByIds(ids) {
-  // seu código aqui
-}
+const animalsByIds = (...ids) =>
+  data.animals.filter(animal => ids.find(id => id === animal.id));
 
-function animalsOlderThan(animal, age) {
-  // seu código aqui
-}
 
-function employeeByName(employeeName) {
-  // seu código aqui
-}
+const animalsOlderThan = (animal, age) =>
+  data.animals
+    .find(element => animal === element.name)
+    .residents.every(obj => obj.age >= age);
 
-function createEmployee(personalInfo, associatedWith) {
-  // seu código aqui
-}
+const employeeByName = function (name) {
+  return name ?
+    data.employees.find(element => element.firstName === name || element.lastName === name)
+    : {};
+};
+
+const createEmployee = (personalInfo, associatedWith) => ({ ...personalInfo, ...associatedWith });
+
+
+const isManager = id => data.employees.some(element => element.managers.includes(id));
```

Comment:
Essa aqui da pra usar um _some_ e um _find_. Legal ver o uso do _include_, nunca ia pensar nisso.
```suggestion
const isManager = id => data.employees.some(({ managers }) => managers.find(index => index === id));
```